yarn-project/foundation/src/serialize/buffer_reader.ts

```diff
@@ -211,9 +211,9 @@ export class BufferReader {
     /**
      * A method that takes a BufferReader instance and returns an instance of the deserialized data type.
      */
-    fromBuffer: (reader: BufferReader) => T;
-  }): T {
-    return deserializer.fromBuffer(this);
+    fromBuffer: (reader: BufferReader, passthrough: any) => T;
```

Comment:
@sklppy88 We discussed about this in a scrum and we would just always couple the logs with the block. This was originally decoupled because we wanted to have the ability to serve the blocks without logs but now that the logs are in tx effects the code is just very not elegant and it seems like a premature optimization.

Reply:
> Should I make a new a issue to nuke the Log Storage ? Or do we want to keep it for the future ?

What do you mean by nuking log storage? If you mean the separate storing of logs in archive then that should be kept as is. If you mean the serialization of blocks with and without logs I've just created an issue for it [here](https://github.com/AztecProtocol/aztec-packages/issues/4635).